core/src/main/java/io/confluent/rest/validation/JacksonMessageBodyProvider.java

```diff
@@ -77,6 +82,8 @@ public Object readFrom(Class<Object> type,
       Throwable cause = e.getCause();
       if (cause instanceof ConstraintViolationException) {
         throw (ConstraintViolationException) cause;
+      }else if(cause instanceof WebApplicationException){
```

Comment:
Why do we need to extract this type of exception now? And do we . Without a test that triggers the specific error, I'm having trouble understanding why we would get a `WebApplicationException` as the cause of the `JsonMappingException` when a key is duplicated. I wouldn't have expected it since `WebApplicationException` is a Jersey error, not a Jackson error, but maybe I'm missing something about the exception handling in this case?

I know I mentioned `WebApplicationException` in the kafka-rest issue, but this wasn't quite what I had in mind. I just meant that somewhere in this `catch` block we might need to handle the specific exception thrown in response to this type of error and convert it to the right type (e.g. that's how invalid constraints found during bean validation are handled by extracting the `ConstraintViolationException`, which is in turn handled elsewhere by an ExceptionMapper to convert it to the appropriate `WebApplicationException`).